packages/UI/src/components/Popover/Popover.stories.tsx

```diff
@@ -1,23 +1,73 @@
-import type { Meta, StoryObj } from '@storybook/react';
-
-import {Popover,  PopoverContent,
-    PopoverTrigger,} from './Popover';
-
-const meta: Meta<typeof Popover> = {
-  component: Popover,
-};
-
-export default meta;
-type Story = StoryObj<typeof Popover>;
-
-/*
- *👇 Render functions are a framework specific feature to allow you control on how the component renders.
- * See https://storybook.js.org/docs/react/api/csf
- * to learn how to use render functions.
- */
-export const Primary: Story = {
-  render: () => <Popover>
-  <PopoverTrigger>Open</PopoverTrigger>
-  <PopoverContent>Place content for the popover here.</PopoverContent>
-</Popover>,
-};
+import type { Meta, StoryObj } from '@storybook/react';
+import Button from '../atoms/Button';
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+  PopoverClose,
+} from './Popover';
+
+const meta: Meta<typeof Popover> = {
+  component: Popover,
+};
+
+export default meta;
+type Story = StoryObj<typeof Popover>;
+
+/*
+ *👇 Render functions are a framework specific feature to allow you control on how the component renders.
+ * See https://storybook.js.org/docs/react/api/csf
+ * to learn how to use render functions.
+ */
+export const Primary: Story = {
+  render: () => (
+    <Popover>
+      <PopoverTrigger>
+        <Button buttonType="primary" buttonSize="standard">
+          Open
+        </Button>
+      </PopoverTrigger>
+      <PopoverContent hasCloseBtn={false}>
+        {' '}
+        Click outside of the popover to close
+      </PopoverContent>
+    </Popover>
+  ),
+};
+
+export const WithClose: Story = {
+  render: () => (
+    <Popover>
+      <PopoverTrigger>
+        <Button buttonType="primary" buttonSize="standard">
+          Open
+        </Button>
+      </PopoverTrigger>
+      <PopoverContent className="w-96"> Click X to close </PopoverContent>
+    </Popover>
+  ),
+};
+
+export const UsingPopoverClose: Story = {
+  render: () => (
+    <Popover>
+      <PopoverTrigger>
+        <Button buttonType="primary" buttonSize="standard">
+          Open
+        </Button>
+      </PopoverTrigger>
+      <PopoverContent
+        hasCloseBtn={false}
+        className="flex w-96 items-center justify-between"
+      >
+        <p>Click button to close</p>
+        <PopoverClose>
+          <Button buttonType="primary" buttonSize="standard">
+            {' '}
+            Close{' '}
+          </Button>
+        </PopoverClose>

```

Comment:
`<PopoverClose>` is actually a button so having nested buttons is a bad practice and makes the component less accessible.

you can pass in `asChild` as boolean prop for that to `<PopoverClose>`